shared/crypto/dkg.go

```diff
@@ -58,6 +76,11 @@ func UnmarshalPubPoly(scheme ThresholdScheme, b []byte) (share.PubPoly, error) {
 	group := scheme.KeyGroup()
 	pointLen := group.PointLen()
 	base := group.Point()
+
+	if len(b) < pointLen {
```

Comment:
I feel like this might not protect from crashes the below 
```
for i := pointLen; i < len(b); i += pointLen {
	p := group.Point()
	err = p.UnmarshalBinary(b[i : i+pointLen])
```

What if `len(b) == 20` and `pointLen == 12` for instance? 

Since the for loop is working as follows:
 -   the init statement: executed before the first iteration
 -   the condition expression: evaluated before every iteration
 -   the post statement: executed at the end of every iteration

My understanding is that we'd have:
1. `i := 12` ✅ 
2. `i < 20`  ✅ 
3. `b[12 : 24]` --> 24 is out of bound => crash ❌